ccf_transaction_processor/Readme.md

```diff
@@ -3,40 +3,175 @@ Licensed under Creative Commons Attribution 4.0 International License
+has been tested only for the CCF virtual enclave mode. The dependencies needed to deploy CCF in an Ubuntu 18.04  VM
+with  virtual enclave mode can be installed by running the following command:
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/getting_started/setup_vm/
+./setup_nodriver.sh
+```
+
+The above script works only when the VM is not behind any proxies. If there are proxies, make the following changes
+before executing the above command:
+
+A. In the setup_nosgx.sh file, modify the command `sudo add-apt-repository ppa:ansible/ansible -y` by adding the `-E` option. The modified command looks like `sudo -E add-apt-repository ppa:ansible/ansible -y`. Add a `sudo` prefix to the `ansible-playbook ccf-dependencies-no-driver.yml` command appearing in the same file.
+
+B. In ccf-dependencies-no-driver.yml (present under the same folder), add the environment option for proxies
+(see https://docs.ansible.com/ansible/latest/user_guide/playbooks_environment.html for reference)
+
+```bash
+environment:
+    http_proxy: <specify-http-proxy-here>
+    https_proxy: <specify-https-proxy-here>
+```
+
+C. Add the repo used by ansible scripts for installing python 3.7
+
+```bash
+sudo add-apt-repository ppa:deadsnakes/ppa
+```
+
+## Build CCF & PDO-TP
+
+CCF uses a combination of cmake& ninja to build the application. Execute the following steps to complete the build process:
+
+A. Add the following lines to the CMakeLists.txt found at $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF.
+    (look for other add_ccf_app in this file, add just above or below this)
+
+```bash
+add_ccf_app(
+pdoenc SRCS ../transaction_processor/pdo_tp.cpp
+../transaction_processor/verify_signatures.cpp
+)
+```
+
+B. Create the build folder
+
+```bash
+mkdir $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+```
+
+C. Set the build flags
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+cmake -GNinja -DCOMPILE_TARGETS=virtual -DBUILD_END_TO_END_TESTS=OFF -DBUILD_SMALLBANK=OFF \
+                -DBUILD_TESTS=OFF-DBUILD_UNIT_TESTS=OFF ..
+```
+
+D. Build using ninja
+
+```bash
+ninja
+```
+
+If the VM has a small memory (< 4GB), it might be required to reduce the parallelism in the build process using the `-j` flag for the `ninja` command, say ninja -j2 or even ninja -j1 (the default seems to be ninja -j4).
+
+If build is successful, the libpdoenc.virtual.so library is created inside the build folder (ignore any other targets that get created)
+
+## Deploy CCF with PDO-TP as the application
+
+The following commands start a single node CCF under the virtual enclave mode.  PDO-TP is the hosted application.
+
+```bash
+cd $PDO_SOURCE_ROOT/ccf_transaction_processor/CCF/build
+python3.7 -m venv env
+source env/bin/activate
+pip install -q -U -r ../tests/requirements.txt
+python ../tests/start_network.py --gov-script ../src/runtime_config/gov.lua  --label pdo_tp -e virtual --package libpdoenc.virtual.so --node <ip-address:6006>
+```
```

Comment:
right now the network uses my personal ssh key at some point. this behavior is not appropriate. if an ssh key is needed for the client it should be created and we should ensure that it is used.